src/engine/wal/manager/mod.rs

```
@@ -123,22 +127,72 @@ where
+        if self.current_segment.is_some() {
+            return Ok(());
+        }
+
         let path = self.current_path();
+        if let Some(parent) = path.parent() {
+            std::fs::create_dir_all(parent).map_err(|e| WALError::wrap(e.to_string()))?;
+        }
 
-        match tokio::fs::OpenOptions::new()
+        let file = OpenOptions::new()
+            .create(true)
```

Clippy (1.96.0)

[warning] file opened with `create`, but `truncate` behavior not defined

(warning)